subprojects/base-services/src/main/java/org/gradle/internal/classloader/CachingClassLoader.java

```diff
@@ -20,13 +20,34 @@
 
 import java.io.Closeable;
 import java.io.IOException;
+import java.lang.reflect.Method;
+import java.lang.reflect.InvocationTargetException;
 import java.util.concurrent.ConcurrentMap;
 
 public class CachingClassLoader extends ClassLoader implements ClassLoaderHierarchy, Closeable {
     private static final Object MISSING_CLASS = new Object();
     private final ConcurrentMap<String, Object> loadedClasses = new MapMaker().weakValues().makeMap();
     private final ClassLoader parent;
 
+    static {
+        /*
+         * This classloader is thread-safe and ClassLoader is parallel capable,
+         * so register as such to reduce contention when running multithreaded builds.
+         * We do so through relfection since Gradle should print error messages when
+         * run with older JRE versions
+        */
+        try {
+            Method m = ClassLoader.class.getMethod("registerAsParallelCapable");
```

Comment:
That won't work because the `registerAsParallelCapable` method is not public but protected. You probably have to use
```java
Method m = ClassLoader.class.getDeclaredMethod("registerAsParallelCapable");
m.setAccessible(true);
```